connaisseur/flask_server.py

```diff
@@ -138,8 +137,14 @@ def __create_logging_msg(msg: str, **kwargs):
     return str({"message": msg, "context": dict(**kwargs)})
 
 
-def __create_json_patch(path: str, index: int, image: Image):
-    return {"op": "replace", "path": path.format(str(index)), "value": str(image)}
+def __update_json_patch(
+    patches: dict, path: str, index: int, image: Image, digest: str
+):
+    if digest:
+        image.set_digest(digest)
```

Comment:
Why this side effect on `image`? (Previously, this was done outside the function.)

Reply:
The static validator return `None` as a trusted digest (if successful), indicating that nothing should change. In this case we don't wanna create any JSON patch and skip ahead. We could set the digest of the image to the trusted one and create patches only if the image has a digest, as we previously did, but then we might overwrite an existing digest with `None`. Connaisseur also checks image references that already have a digest.

Reply:
My question was not about the `if`, but about the side effect on `image`, namely that we're modifying the `image` object that is given as an argument to `__update_json_patch`. This function name doesn't sound like it would modify any of its arguments.

Why not do the `if` and `image.set_digest` outside of `__update_json_patch`, similar to what was done before?